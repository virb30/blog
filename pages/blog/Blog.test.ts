import Vuex from 'vuex';
import VueMeta from "vue-meta";
import { createLocalVue, mount } from "@vue/test-utils";
import Index from './index.vue';

const localVue = createLocalVue();
localVue.use(VueMeta, { keyName: 'head' });
localVue.use(Vuex);

describe('Blog Page', () => {

	let store: any;
	let config: any;

	beforeEach(() => {

		const mockGetter = jest.fn();
		mockGetter.mockImplementationOnce(() => []);

		const mockFetchPosts = jest.fn();
		mockFetchPosts.mockImplementationOnce(() => []);

		store = new Vuex.Store({
			modules: {
				Blog: {
					namespaced: true,
					state: {
						posts: [],
						pagination: {},
					},
					getters: {
						last3Posts: mockGetter,
					},
					actions: {
						fetchPosts: mockFetchPosts,
					}
				}
			}
		});

		config = {
			localVue,
			stubs: {
				Article: true,
				Pagination: true,
			},
			store,
		};

	});

	it('should render childs', () => {
		const wrapper = mount(Index, config);

		expect(wrapper.html()).toMatchSnapshot();
	});

	describe('Meta info', () => {
		it('should have a meta title', () => {
			const wrapper = mount(Index, config);

			const expected = 'Blog | viniboscoa.dev';

			const meta = wrapper.vm.$meta().refresh();

			expect(meta.metaInfo.title).toBe(expected);
		});
	});
});
