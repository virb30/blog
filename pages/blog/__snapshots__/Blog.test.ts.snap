// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Page should render childs 1`] = `
"<div>
  <main class=\\"m-auto max-w-6xl p-4\\">
    <header class=\\"mt-12 mb-8\\">
      <h1 class=\\"font-bold text-2xl mb-3\\">
        Blog
      </h1>
      <p class=\\"font-light text-xl\\">
        Conte√∫do sobre o Desenvolvimento Web e Data Science
      </p>
    </header>
    <div class=\\"grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3 flex-col sm:flex-row lg:flex-row w-full flex-wrap\\"></div>
    <pagination-stub pagination=\\"[object Object]\\"></pagination-stub>
  </main>
</div>"
`;
